lisa/tests/kernel/scheduler/misfit.py

```diff
@@ -127,16 +127,7 @@ def __init__(self, res_dir, plat_info, rtapp_profile):
         super().__init__(res_dir, plat_info, rtapp_profile)
 
         cpu_capacities = plat_info['cpu-capacities']
-
-        cpu_classes = {}
-        for cpu, capacity in cpu_capacities.items():
-            if capacity not in cpu_classes.keys():
-                cpu_classes[capacity] = []
-
-            cpu_classes[capacity].append(cpu)
-
-        capacities = sorted(cpu_classes.keys())
-        self.cpu_classes = [cpu_classes[capacity] for capacity in capacities]
+        self.cpu_classes = plat_info['capacity-classes']
```

Comment:
may be better to refer to self.plat_info where it is actually needed. That allows easier refactoring (the code just depends on a `plat_info` to work, instead of a specific class), and allows better tracking what data is used as the input of each test by looking at the debug log (which logs access to `plat_info`).

Reply:
I feel much more comfortable storing this value into a variable, especially since it's only ever used in this method - I now see it's only used in the `__init__()` so I could make it a local variable instead of an instance attribute .

Reply:
right, if it's just some local use then there is no issue. What we just want to avoid is ending with quite generic code, that ends up seemingly deeply tied to a class where in fact, it could easily be extracted and put in a helper (or a base class) if needed.